frontend/src/pages/LogIn/index.tsx

```diff
@@ -77,38 +61,53 @@ const LogIn: React.FC = () => {
           <Stack
             component="form"
             direction="column"
-            spacing={1.5}
             sx={(theme) => ({
               marginTop: theme.spacing(2),
               marginBottom: theme.spacing(2),
             })}
-            onSubmit={handleLogIn}
+            onSubmit={handleSubmit((data) => login(data.email, data.password))}
             noValidate
           >
-            <CustomTextField
+            <TextField
               label="Email"
-              size="small"
               required
-              emptyField={emptyFields.email}
-              validator={emailValidator}
-              onChange={(value, isValid) =>
-                handleInputChange("email", value, isValid)
-              }
+              fullWidth
+              margin="normal"
+              type="email"
+              {...register("email", { validate: { emailValidator } })}
+              error={!!errors.email}
+              helperText={errors.email?.message}
             />
-            <CustomTextField
+            <TextField
               label="Password"
-              size="small"
               required
-              emptyField={emptyFields.password}
-              onChange={(value, isValid) =>
-                handleInputChange("password", value, isValid)
-              }
-              isPasswordField
+              fullWidth
+              margin="normal"
+              {...register("password", { validate: { passwordValidator } })}
+              error={!!errors.password}
+              helperText={errors.password?.message}
+              type={showPassword ? "text" : "password"}
+              slotProps={{
+                input: {
+                  endAdornment: (
+                    <InputAdornment position="end">
+                      <IconButton
+                        onClick={() => setShowPassword((prev) => !prev)}
```

Comment:
For clarity and consistency with the implementation in the sign up page, maybe this instead?

`onClick={() => setShowPassword(!showPassword)}`

Reply:
i think it would be better to do `setShowPassword((prev) => !prev))` since the update depends on the previous state.

https://react.dev/reference/react/useState#updating-state-based-on-the-previous-state

Reply:
Ah okay that makes sense, then we should probably change the implementation in the sign up page